### PR TITLE
Add Contribution Section

### PR DESCRIPTION
**Pull request by kgal-pan**
## Status
Ready

## Related Issues
Related: https://jira-hq.paloaltonetworks.local/browse/CIAC-8359

## Description
This PR includes the following changes:

* Restructured Contributing section by adding sub sections about licensing, prerequisites, and the many ways developers can contribute.
* Remove Python 2-based image mentions.
* Some minor grammar, backticks added.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -7,87 +7,108 @@ This repository's `master` branch tracks images pushed to the official Demisto D
...
+* [**Create a new Docker Image**](#create-a-new-docker-image) for use in an [XSOAR/XSIAM integration or script](https://xsoar.pan.dev/docs/integrations/docker#scriptintegration-configuration).
+
+* **Update existing Docker Images:** If you encounter issues or security concerns in our `Dockerfiles`, feel free to open a Pull Request. 
+
+* **Enhancements:** Feel free to submit Pull Requests with your proposed changes. Enhancements can include optimizations, helper scripts, or improved configurations.
+
+* **Documentation:** Documentation is crucial. If you find gaps or errors in our documentation, submit PRs to help us improve it. You can also suggest clarifications or additions to make it more user-friendly.
+
+### Licenses
+In order to conntribute, we will need all contributors to sign a contributor license agreement (CLA). By signing a contributor license agreement, we ensure that the community is free to use your contributions.
```
Comment (JudahSchwartz) [inline README.md:23]:
```suggestion
All contributors are required to sign a contributor license agreement (CLA) to ensure that the community is free to use your contributions.
```

File: README.md [hunk 2]
```diff
@@ -7,87 +7,108 @@ This repository's `master` branch tracks images pushed to the official Demisto D
...
-  * Mac: use brew to install (more info at: https://docs.brew.sh/Homebrew-and-Python): `brew install python3` and then: `brew install python@2`
-* Or install pyenv (recommended for managing multiple python versions):
-    * Mac: `brew install pyenv` . Make sure to run then: `pyenv init` and follow instructions to add to either `~/.zshrc` or `~/.bash_profile` depending on you shell.
-    * Other: see https://github.com/pyenv/pyenv-installer 
-* Install pipenv globally using: `pip install pipenv`
-* Install requests globally: `pip install requests`
+#### Prequisites
+Make sure you meet the following prerequisites:
+* An active GitHub account.
+* A [clone](https://docs.github.com/en/repositories/creating-and-managing-repositories/cloning-a-repository) or a [fork](https://docs.github.com/en/get-started/quickstart/fork-a-repo) of the repository. The main difference is that if the contribution is done from a fork, the development Docker image created during the Pull Request build will not be deployed to DockerHub whereas opening a Pull Request from a clone will deploy the image.
```
Comment (JudahSchwartz) [inline README.md:36]: I think this isnt correct. The only way to contribute is from a fork. Cloning the repository just means to make a local copy, but the commits are done in the same repo.
Contributors arent allowed to write to our repo at all, so currently there isnt a way to deploy their changes to our dockerhub account (we have a task open for this, lets discuss)

The current solution is to follow the build output which provides a command to run in order to load the image locally. I would add this note here

File: README.md [hunk 3]
```diff
@@ -188,11 +207,12 @@ If you want to use poetry, make sure you have poetry installed by running `poetr
 * Make sure to commit both `pyproject.toml` and `poetry.lock` files
 * To see the locked dependencies run: `poetry export -f requirements.txt --output requirements.txt --without-hashes` 
 
-### Pipenv quick start:
-If you want to use pipenv manually make sure you first have the pre-requisites installed as specified in [Getting Started](#getting-started). Then follow:
+### `pipenv` quick start:
```
Comment (JudahSchwartz) [inline README.md:210]: can we add a note in here that poetry is preferred over pipenv?

File: README.md [hunk 4]
```diff
@@ -7,87 +7,108 @@ This repository's `master` branch tracks images pushed to the official Demisto D
...
+
+  usage: create_new_docker_image.py [-h] [-t {python,powershell}]
+                                    [-p {two,three}] [-l {alpine,debian,ubuntu}]
+                                    [--pkg PKG]
+                                    name
 
-## Building a docker image locally
+  Create a new docker image
+
+  positional arguments:
```
Comment (JudahSchwartz) [inline README.md:59]: can we put this --help block inside a code block?

File: README.md [hunk 5]
```diff
@@ -7,87 +7,108 @@ This repository's `master` branch tracks images pushed to the official Demisto D
...
-When opening a new pull request, a bot will evaluate whether you have signed the CLA. If required, the bot will comment on the pull request, including a link to accept the agreement. The CLA document is also available for review as a [PDF](https://github.com/demisto/content/blob/master/docs/cla.pdf).
+* [**Create a new Docker Image**](#create-a-new-docker-image) for use in an [XSOAR/XSIAM integration or script](https://xsoar.pan.dev/docs/integrations/docker#scriptintegration-configuration).
+
+* **Update existing Docker Images:** If you encounter issues or security concerns in our `Dockerfiles`, feel free to open a Pull Request. 
+
+* **Enhancements:** Feel free to submit Pull Requests with your proposed changes. Enhancements can include optimizations, helper scripts, or improved configurations.
+
+* **Documentation:** Documentation is crucial. If you find gaps or errors in our documentation, submit PRs to help us improve it. You can also suggest clarifications or additions to make it more user-friendly.
+
+### Licenses
```
Comment (JudahSchwartz) [inline README.md:22]:
```suggestion
### CLA Licenses
```

File: README.md [hunk 6]
```diff
@@ -7,87 +7,108 @@ This repository's `master` branch tracks images pushed to the official Demisto D
...
+You can contribute in the following ways:
 
-When opening a new pull request, a bot will evaluate whether you have signed the CLA. If required, the bot will comment on the pull request, including a link to accept the agreement. The CLA document is also available for review as a [PDF](https://github.com/demisto/content/blob/master/docs/cla.pdf).
+* [**Create a new Docker Image**](#create-a-new-docker-image) for use in an [XSOAR/XSIAM integration or script](https://xsoar.pan.dev/docs/integrations/docker#scriptintegration-configuration).
+
+* **Update existing Docker Images:** If you encounter issues or security concerns in our `Dockerfiles`, feel free to open a Pull Request. 
+
+* **Enhancements:** Feel free to submit Pull Requests with your proposed changes. Enhancements can include optimizations, helper scripts, or improved configurations.
+
+* **Documentation:** Documentation is crucial. If you find gaps or errors in our documentation, submit PRs to help us improve it. You can also suggest clarifications or additions to make it more user-friendly.
```
Comment (JudahSchwartz) [inline README.md:20]:
```suggestion
* **Update existing Docker Images:** to fix issues or security concerns in our `Dockerfiles`. 

* **Enhancements:**  such as optimizations, helper scripts, or improved configurations.

* **Documentation:** Documentation is crucial. If you find gaps or errors in our documentation, please help us improve it. You can suggest clarifications or additions to make it more user-friendly.
```

**Review by JudahSchwartz — COMMENTED** [review]
Thanks so much for helping improve the docs!

File: README.md [hunk 7]
```diff
@@ -7,87 +7,115 @@ This repository's `master` branch tracks images pushed to the official Demisto D
...
-## Getting Started
-Each docker image is managed in its own directory. The directory should be named the same as the image name (without the organization prefix). If needed, we prefer using a dash (`-`) as a separator in the name. All image directories are located under the `docker` dir.
+After opening a Pull Request, and in order for the reviewer to understand the context, make sure to link to the corresponding Pull Request from the [Content](https://github.com/demisto/content) repo where this Docker image will be used.
 
-The directory should contain one Dockerfile which will be used for building the docker image. Each image when it is built is tagged with the commit hash and version. 
 
-The script `docker/build_docker.sh` is used to build all modified docker images. The script detects modified directories by comparing against origin/master if on a branch or if on master by using the CIRCLE_COMPARE_URL environment variable to obtain the commit range of the current build.
+### How to Contribute
```
Comment (JudahSchwartz) [inline README.md:31]: why have contributing and how to contribute as 2 seperate sections?

File: README.md [hunk 8]
```diff
@@ -7,87 +7,115 @@ This repository's `master` branch tracks images pushed to the official Demisto D
...
+                          For example: ldap3. We use kebab-case naming
+                          convention.
+
+  optional arguments:
+    -h, --help            show this help message and exit
+    -t {python,powershell}, --type {python,powershell}
+                          Specify type of image to create (default: python)
+    -p {two,three}, --python {two,three}
+                          Specify python version to use (default: three)
+    -l {alpine,debian,ubuntu}, --linux {alpine,debian,ubuntu}
```
[the remark this answers is not in the document]
Comment (kgal-pan) [reply README.md]: https://github.com/demisto/dockerfiles/pull/20820/commits/71ed1cd8363e9a8dce27d26501ea0cab1ff59751

File: README.md [hunk 9]
```diff
@@ -7,87 +7,115 @@ This repository's `master` branch tracks images pushed to the official Demisto D
...
-  -p {two,three}, --python {two,three}
-                        Specify python version to use (default: three)
-  -l {alpine,debian,ubuntu}, --linux {alpine,debian,ubuntu}
-                        Specify linux distro to use (default: alpine)
-  --pkg PKG             Specify a package/module to install. Can be specified
-                        multiple times. Each package needs to be specified
-                        with --pkg. For example: --pkg google-cloud-storage
-                        --pkg oath2client (default: None)
-```
+Once all prerequisites have been met, create a new branch to hold the proposed work:
```
[the remark this answers is not in the document]
Comment (kgal-pan) [reply README.md]: https://github.com/demisto/dockerfiles/pull/20820/commits/71ed1cd8363e9a8dce27d26501ea0cab1ff59751

File: README.md (hunk 7 above)
Comment (kgal-pan) [reply README.md]: Addressed in https://github.com/demisto/dockerfiles/pull/20820/commits/5d530e9e0adcd77c6df52a7cdf1fe3692d1f8d37.

**Pull request merged**